File: src/main/java/net/atos/client/zgw/drc/model/ObjectInformatieobjectListParameters.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.drc.model;

import java.net.URI;

import jakarta.ws.rs.QueryParam;

public class ObjectInformatieobjectListParameters {

  /**
   * URL-referentie naar het gerelateerde OBJECT (in deze of een andere API).
   */
  @QueryParam("object")
  private URI object;

  /**
   * URL-referentie naar het INFORMATIEOBJECT.
   */
  @QueryParam("informatieobject")
  private URI informatieobject;

  public ObjectInformatieobjectListParameters() {}

  public URI getObject() {
    return object;
  }

  public void setObject(URI object) {
    this.object = object;
  }

  public URI getInformatieobject() {
    return informatieobject;
  }

  public void setInformatieobject(URI informatieobject) {
    this.informatieobject = informatieobject;
  }
}

File: src/main/kotlin/nl/info/client/zgw/drc/model/ObjectInformatieobjectListParameters.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */

package nl.info.client.zgw.drc.model

import jakarta.ws.rs.QueryParam
import nl.info.zac.util.AllOpen
import nl.info.zac.util.NoArgConstructor
import java.net.URI

@NoArgConstructor
@AllOpen
class ObjectInformatieobjectListParameters {

    /**
     * URL-referentie naar het gerelateerde OBJECT (in deze of een andere API).
     */
    @QueryParam("object")
    var objectUri: URI? = null

    /**
     * URL-referentie naar het INFORMATIEOBJECT.
     */
    @QueryParam("informatieobject")
    var informatieobject: URI? = null
}
